modules/social_features/social_event/social_event.module

```diff
@@ -638,3 +724,161 @@ function social_event_get_current_event() {
 
   return NULL;
 }
+
+/**
+ * Add conditional field visibility triggers.
+ *
+ * @param array $form
+ *   The form that needs to be altered.
+ * @param string $field_name
+ *   The field that acts as a trigger.
+ * @param array $fields
+ *   The field that should show upon trigger.
+ * @param string $state
+ *   Which state do you want for the form field. Defaults to visibility.
+ */
+function social_event_field_visibility(array &$form, $field_name, array $fields, $state = 'visible') {
+  // Todo:: Refactor outside of an alter when going to React.
+  // Hide fields if event enrollment is disabled for event.
+  $trigger = [
+    ':input[name="' . $field_name . '[value]"]' => ['checked' => TRUE],
+  ];
+
+  foreach ($fields as $field) {
+    $form[$field]['#states'] = [
+      $state => $trigger,
+    ];
+  }
+}
+
+/**
+ * Determine if the user is an owner or organizer of a specific event node.
+ *
+ * @param \Drupal\node\NodeInterface|null $node
+ *   The (optional) event node.
+ * @param \Drupal\Core\Session\AccountInterface|null $current_user
+ *   The (optional) user.
+ *
+ * @return bool
+ *   Return TRUE if the user is the owner or organizer of this event.
+ */
+function social_event_owner_or_organizer(NodeInterface $node = NULL, AccountInterface $current_user = NULL) {
```

Comment:
@frankgraave this is better suited for a `service` instead of in a module file.

Reply:
Alright, will move this towards a service. 👍